app/src/main/java/com/udacity/nanodegree/blooddonation/ui/home/HomeActivity.java

```diff
@@ -15,4 +20,25 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_home);
     }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        getMenuInflater().inflate(R.menu.menu_main , menu);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        int itemId = item.getItemId();
+        switch(itemId){
+            case R.id.menu_sign_out :
+                FirebaseAuth.getInstance().signOut();
+                if (FirebaseAuth.getInstance().getCurrentUser() == null) {
```

Comment:
No need of this extra check after singout.

Reply:
Wait for the UI or UX to come for signout option, the user should be able to signout from anywhere in the app. Just, not from here.

@KamalMrock  UI, UX for the same

Reply:
Copy that